### PR TITLE
**WIP** TypeScript setup

### PR DESCRIPTION
**Pull request by MehulKChaudhari**
This PR contains configurations of TypeScript. 

# Do not merge this `PR` is under progress :warning:

### CODE REVIEW

File: utils/logger.js [hunk 1]
```diff
@@ -29,19 +31,20 @@ const logger = new winston.createLogger({ // eslint-disable-line new-cap
...
+    ...(config.get('enableFileLogs') ? [new _transports.File(options.file)] : []),
+    ...(config.get('enableConsoleLogs') ? [new _transports.Console(options.console)] : [])
   ],
 
   exitOnError: false // do not exit on handled exceptions
 })
 
 // create a stream object with a 'write' function that will be used by `morgan`
 logger.stream = {
+  // @ts-ignore
```
Comment (rohan09-raj) [inline utils/logger.js:43]: Is this because you were getting errors in the initial setup ??

**Review by rohan09-raj — CHANGES_REQUESTED** [review]
Please resolve the conflicts in package.json.

File: utils/logger.js (hunk 1 above)
Comment (MehulKChaudhari) [reply utils/logger.js:43]: Yes and no, I used `@ts-ignore` when I was checking the setup and converted this file to ts. But as this file is now a js file we can remove it. But I kept it so when we change it to TS we don't get a lot of errors and later fix the type.

**Comment by MehulKChaudhari** [comment]
> Please resolve the conflicts in package.json.

Yes, this conflict because the develop branch now uses yarn, and this branch is old so it uses npm.

**Comment by rohan09-raj** [comment]
> > Please resolve the conflicts in package.json.
> 
> Yes, this conflict because the develop branch now uses yarn, and this branch is old so it uses npm.

Ok got it. Have you done a rebase of your branch with develop ??

**Comment by MehulKChaudhari** [comment]
> > > Please resolve the conflicts in package.json.
> > 
> > 
> > Yes, this conflict because the develop branch now uses yarn, and this branch is old so it uses npm.
> 
> Ok got it. Have you done a rebase of your branch with develop ??

doing that bro

File: app.js [hunk 2]
```diff
@@ -31,4 +32,5 @@ app.use(function (err, req, res, next) {
   }
 })
 
-module.exports = app
+// module.exports = app
+export default app
```
Comment (akshay1502) [inline app.js:36]: can we stick to either es5 or es6 way import/export?

File: server.js [hunk 3]
```diff
@@ -38,30 +39,33 @@ server.listen(port)
...
+// @ts-ignore
 function onError (error) {
   if (error.syscall !== 'listen') {
     throw error
   }
 
-  const bind = typeof port === 'string'
-    ? 'Pipe ' + port
-    : 'Port ' + port
+  const bind = typeof port === 'string' ? 'Pipe ' + port : 'Port ' + port
```
Comment (akshay1502) [inline server.js:54]: could you tell what could be the possible values of port?

Comment (MehulKChaudhari) [reply server.js:54]: Hi Akshay, I don't have any idea about this. In my PR this line spacing got changed by prettier so its says I have made those changes but I haven't.

File: app.js (hunk 2 above)
Comment (MehulKChaudhari) [reply app.js:36]: Yes sure.

File: server.js (hunk 3 above)
Comment (akshay1502) [reply server.js:54]: Fine 😅.

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
+  "compilerOptions": {
+    "target": "es5",
+    "module": "commonjs",
+    "lib": ["esnext"],
+    "allowJs": true,
+    "outDir": "build",
+    "rootDir": "./",
+    "strict": true,
```
Comment (swarajpure) [inline tsconfig.json:9]: Is `strict: true` causing the errors?

File: controllers/testTypescript.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+export const testTypescript = async (req: any, res: any) => {
+  return res.json({
+    message: "Working perffectly fine.",
```
Comment (harshith-venkatesh) [inline controllers/testTypescript.ts:3]: Spelling mistake 🙃

Comment (harshith-venkatesh) [reply controllers/testTypescript.ts:3]: Or was it intentional😅

File: routes/index.js [hunk 6]
```diff
@@ -15,5 +15,6 @@ app.use("/trade", require("./trading.js"));
 app.use("/users", require("./users.js"));
 app.use("/wallet", require("./wallets.js"));
 app.use("/featureFlags", require("./featureFlags"));
+app.use("/mehul", require("./testTypescript"));
```
Comment (harshith-venkatesh) [inline routes/index.js:18]: Maybe test-typescript would be a better name here?
If you wish You can add a comment as /** author: mehul **/

File: routes/testTypescript.ts [hunk 7]
```diff
@@ -0,0 +1,7 @@
+const express = require("express");
+const router = express.Router();
+const member = require("../controllers/testTypescript");
+
+router.get("/mehul", member.testTypescript);
```
Comment (harshith-venkatesh) [inline routes/testTypescript.ts:5]: What is the purpose of member.testTypescript here?

**Review by harshith-venkatesh — CHANGES_REQUESTED** [review]

File: controllers/testTypescript.ts (hunk 5 above)
Comment (MehulKChaudhari) [reply controllers/testTypescript.ts:3]: Testing file will be removed before merging to develop

File: routes/index.js (hunk 6 above)
Comment (MehulKChaudhari) [reply routes/index.js:18]: Testing code will be removed before merging to develop

File: package.json [hunk 8]
```diff
@@ -3,8 +3,9 @@
   "version": "1.0.0",
   "private": true,
   "scripts": {
-    "start": "node server.js",
-    "dev": "cross-env NODE_ENV=development nodemon server.js",
+    "start": "node ./dist/server.js",
```
Comment (shubham-y) [inline package.json:6]: Can we use ts-node instead of node?

File: tsconfig.json [hunk 9]
```diff
@@ -0,0 +1,19 @@
+{
+  "compilerOptions": {
+    "target": "es5",
+    "module": "commonjs",
+    "lib": ["esnext"],
+    "allowJs": true,
+    "outDir": "build",
```
Comment (shubham-y) [inline tsconfig.json:7]: Should'nt the **outDir** be `dist`?
As the path is [`./dist/server.js`](https://github.com/Real-Dev-Squad/website-backend/blob/42a16c6f1fc88f8039b4bd2ab0b0fb6c550a8999/package.json#L6) in package.json.

Comment (MehulKChaudhari) [reply tsconfig.json:7]: Yes yes, my bad here. Thanks for pointing this out

File: package.json (hunk 8 above)
Comment (MehulKChaudhari) [reply package.json:6]: I don't think so we need to use `ts-node` as per my knowledge but if there are some extra benefits of using `ts-node` we can discuss this.

**Pull request closed without merging**